src_c/context.c

```diff
@@ -26,9 +26,81 @@ pg_context_get_pref_path(PyObject *self, PyObject *args, PyObject *kwargs)
     return ret;
 }
 
+static PyObject *
+pg_context_get_pref_locales(PyObject *self, PyObject *_null)
+{
+    PyObject *ret_list = PyList_New(0);
+    if (!ret_list) {
+        return NULL;
+    }
+
+#if SDL_VERSION_ATLEAST(2, 0, 14)
+    PyObject *dict, *val = NULL;
+    SDL_Locale *locales = SDL_GetPreferredLocales();
+    if (!locales) {
+        /* Return an empty list if SDL function does not return any useful
+         * information */
+        return ret_list;
+    }
+
+    SDL_Locale *current_locale = locales;
+    while (current_locale->language) {
```

Comment:
Maybe add a comment explaining the list is terminated by a locale with a NULL language field. Just since this is kind of a strange pattern without `context`.

Reply:
yeah good idea